### DISCUSSION TITLE
how to get each row number in limereport? using QTableWidget as datasource

### DISCUSSION THREAD

**Discussion by proton07** [fralx/LimeReport#402] [Q&A] [answered]
Hi this is the best reporting engine ever thank you really for your great work
, i have a question , so i have Qtablewidget which contain a 3 columns 
i want to get each row number when reporting using limereport
is there any way to do that!
for example i want those numbers on limereport
https://i.imgur.com/uUJmXqE.jpg

**Comment by fralx** [comment, marked as answer]
Hi! LimeReport has embedded function: line(Databand name). You can use it like this: ```$S{line("DataBand1")} ```

**Reply by proton07** [reply]
Thank You so much , You are The best bro <3
